src/controllers/hid/hidcontroller.cpp

```diff
@@ -208,8 +220,22 @@ bool HidController::poll() {
             return false;
         } else if (bytesRead == 0) {
             // No packet was available to be read
+            for (int rptIdx = 0; rptIdx <= kMaxHidInputReportId; rptIdx++) {
+                if (inputReportCounter[rptIdx] > 1) {
+                    qCWarning(m_logOutput)
+                            << getName() << ": Read "
+                            << inputReportCounter[rptIdx]
+                            << " HID input reports with Report ID " << rptIdx
+                            << "in the same polling interval. This indicates a "
+                               "performance problem with controller mapping or "
+                               "a too slow polling interval.";
+                }
+            }
             return true;
         }
+        inputReportCounter[m_pPollData
+                        [m_pollingBufferIndex]
+                        [0]]++; // Count reports read in this while loop per ID
```

Comment:
This one is hard to read, can we help clang format a bit? 

```suggestion
        // Count reports read in this while loop per ID
        unsigned char rptIdx = m_pPollData[m_pollingBufferIndex][0];
        inputReportCounter[rptIdx]++;
```